App/Reducers/WeatherReducer.js

```diff
@@ -0,0 +1,43 @@
+import Types from '../Actions/Types'
+import Immutable from 'seamless-immutable'
+import createReducer from './CreateReducer'
+
+export const INITIAL_STATE = Immutable({
+  temperature: null,
+  fetching: null,
+  error: null,
+  city: null
+})
+
+// request temp
+const request = (state, action) =>
+  state.merge({
```

Comment:
I'll be honest, I'm not in love with seamless-immutable `.merge` because it gives the connotation that a merge is happening, and not a brand new object.   I just have to say this complaint to assuage my cognitive dissonance.